code/modules/power/singularity/emitter.dm

```diff
@@ -288,4 +335,57 @@
 	return burst_delay
 
 /obj/machinery/power/emitter/proc/get_emitter_beam()
-	return new /obj/item/projectile/beam/emitter(get_turf(src))
\ No newline at end of file
+	return new /obj/item/projectile/beam/emitter(get_turf(src))
+
+
+//Safety mechanisms - 2023-03-07/2023-03-23
+//Added due to in-character Union actions (and multiple disintegrated arms)
+//Summary: when the emitter is dormant, a targeting laser can be present (toggled with alt-click).
+//When the emitter is firing, the targeting laser WILL be present.
+//Multitools temporarily disable the targeting software. Emags permanently disable the targeting software... and scramble the fire delays.
+/obj/machinery/power/emitter/AltClick(mob/user)
+	. = ..()
+	try_toggle_safety(user)
+
+/obj/machinery/power/emitter/proc/try_toggle_safety(mob/user) //Enable or disable the targeting laser when it's dormant
+	if(state != STATE_WELDED)
+		to_chat(user, "<span class='notice'>You can't adjust the emitter's targeting laser without securing it first.</span>")
+		return
+	if(active)
+		to_chat(user, "<span class='danger'>You can't adjust \the [src]'s targeting laser while it's firing!</span>")
+		return
+	if(safety_override)
+		to_chat(user, "<span class='danger'>\The [src]'s targeting laser has been manually disabled!</span>")
+		return
+	if(emagged) //You weren't standing in front of it, right?
+		to_chat(user, "<span class='warning'>The targeting laser on [src] short-circuits!</span>")
+		activate(user)
+		return
+	to_chat(user, "You adjust \the [src]'s targeting laser.") //give feedback to user!
+	toggle_safety()
+	return
+
+/obj/machinery/power/emitter/proc/toggle_safety()
+	if(!emagged && (powernet || !active_power_usage) && (!safety_override || active)) //Emagged, overridden or depowered = no safety. Otherwise, toggle.
+		safety = !safety
+	else
+		safety = FALSE //just to make super sure
+
+/obj/machinery/power/emitter/proc/toggle_safety_override(mob/user) //Prevent the targeting laser from showing up at all
+	if(state != STATE_WELDED)
+		to_chat(user, "<span class='notice'>You can't adjust the emitter's targeting software without securing it first.</span>")
+		return
+	if(active)
+		to_chat(user, "<span class='danger'>You can't adjust \the [src]'s targeting software while it's firing!</span>")
+		return
+	if(emagged) //You weren't standing in front of it, right?
+		to_chat(user, "<span class='warning'>The targeting laser on [src] short-circuits!</span>")
+		activate(user)
+		return
+
+	safety_override = !safety_override
+	if(safety_override)
+		to_chat(user, "You disable [src]'s targeting software. It was just getting in the way anyway.")
+		safety = FALSE
+	else
+		to_chat(user, "In a brief moment of wisdom, you reenable [src]'s targeting software.")
```

Comment:
```suggestion
/obj/machinery/power/emitter/proc/update_safety()
	safety = !emagged && (powernet || !active_power_usage) && (!safety_override || active) //Emagged, overridden or depowered = no safety. Otherwise, toggle.

/obj/machinery/power/emitter/proc/can_toggle_safety(mob/user)
	if(state != STATE_WELDED)
		to_chat(user, "<span class='notice'>You can't adjust the emitter's targeting laser without securing it first.</span>")
		return FALSE
	if(active)
		to_chat(user, "<span class='danger'>You can't adjust \the [src]'s targeting laser while it's firing!</span>")
		return FALSE
	if(safety_override)
		to_chat(user, "<span class='danger'>\The [src]'s targeting laser has been manually disabled!</span>")
		return FALSE
	if(emagged) //You weren't standing in front of it, right?
		to_chat(user, "<span class='warning'>The targeting laser on [src] short-circuits!</span>")
		activate(user)
		return FALSE
	return TRUE

/obj/machinery/power/emitter/proc/try_toggle_safety(mob/user) //Enable or disable the targeting laser when it's dormant
	if(!can_toggle_safety(user))
		return
	to_chat(user, "You adjust \the [src]'s targeting laser.") //give feedback to user!
	safety = !safety
	return

/obj/machinery/power/emitter/proc/toggle_safety_override(mob/user) //Prevent the targeting laser from showing up at all
	if(!can_toggle_safety(user))
		return
	safety_override = !safety_override
	if(safety_override)
		to_chat(user, "You disable [src]'s targeting software. It was just getting in the way anyway.")
	else
		to_chat(user, "In a brief moment of wisdom, you reenable [src]'s targeting software.")
	update_safety()
```